### DISCUSSION TITLE
Hello World!

### DISCUSSION THREAD

**Discussion by ianthomaz** [choraria/workspacedevs.com#28] [General]
Hello you all ! 

I've a lot to Thank already to @benlcollins , @schoraria911 , Martin (from Pulse) for [this article.](https://pulse.appsscript.info/p/2021/01/workspace-devs-directory/) 

I've started learning code 3 months ago.
Must say Ben's job is amazing. Really reachable to keep up.

But...
I'm getting to a curve point that need's some help and pushup.

Wile researching i'm trying to create a first new DataBase.
Having: Clients / Projects / Budgets / Expenses / Suppliers B2B / Suppliers P2C / 2 Stock Inventory / 2 Completely types of report (financial / Media).

It's been a lot of work. I've no Idea about SQL, But i've small data, it can be done around GoogleSheet.
My Client List is Small, does not need a faster update form as my expenses Log or Inventory In/Out .
There fore, once a day I get something done.

Sometimes a GoogleForm, some I try a WebApp from.

What happens is that i loose a lot of time jut to understand basics.
How to get something, or properly do a ForEach.

Is this a good place to find some help?
I'm also looking for a freelancer to help out on this dev so it goes a bit better (and faster) by it's beginning.
Where should I announce ? Is there a Brazilian community ? 

Tnks a lot!
stay safe!

Ian

**Comment by choraria** [comment]
Hey Ian,

Thanks for reaching out and glad to hear that you've started your journey with Apps Script 💪🏽

I can totally relate towards a lot of time being consumed just to understand the syntax and the basics of the function / method we may want to use but quite honestly, that's been the best part of the journey to me 😄  we get so much to learn and discover along the way.

Here's some pointers that might address some of the questions proposed -
- I learnt about forEach using [this documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) but there's also a nice video attached to this [article from freecodecamp](https://www.freecodecamp.org/news/javascript-foreach-how-to-loop-through-an-array-in-js/)
- In case you already have some code and need help to make it work, [Stack Overflow](https://stackoverflow.com/questions/tagged/google-apps-script) would be your best bet along with the Google Apps Script Community [google groups](https://developers.google.com/apps-script/community)
- I started searching for "consultant" & "freelancer" on [workspacedevs.com](https://workspacedevs.com/) and found these folks who you could try reaching out to individually

    <details>
    <summary>screenshot</summary>
    
    ![Screenshot 2021-01-20 at 12 41 25 PM](https://user-images.githubusercontent.com/37455462/105139936-f6a2ac00-5b1c-11eb-8e14-115157205b2c.png)
    
    </details>

- I don't know of a specific Brazilian community but most of us are also fairly active on Twitter. Feel free to post a link with your requirements and maybe even use the hashtags like #GoogleAppsScript or #AppsScript or even tag [@AppsScriptInfo](https://twitter.com/AppsScriptInfo) ([Martin Hawksey](https://twitter.com/mhawksey) runs this channel and he generally retweets these kinds of requests ☺️)

Hope these help and I'll look forward to what you end up building!

Best,
Sourabh
[@schoraria911](https://twitter.com/schoraria911)

**Comment by ianthomaz** [comment]
Hi @schoraria911 !!!

Tnks a Lot for this detailed and fast reply.
I've took some time today to see more about the community.

Got on the G-Group on email, and loved all new references for help you brought.
I'm not Putting my codes on Stack at the moment, but i've used it a lot to help me complete some lines of code.

I do hope to find someone that could fit on budget we have over here.
Also to make my learning a bit better.

My next attempt will be completing these 2 Udemy courses I got for just a few U$.
https://www.udemy.com/share/103QWKAEAbdFlbQ3UJ/
https://www.udemy.com/share/1040s2AEAbdFlbQ3UJ/

Tnks a lot! 
I'll keep track of the progress over here as well.

Ian
